src/base/decoder/baseDecoder.ts

```diff
@@ -0,0 +1,33 @@
+import { ContractABI } from '../iface';
+import { BaseFunctionCallExplanation, BaseMethodIdMapping } from './iface';
+
+// import { MethodContainerMap, MethodManager } from '../method/manager';
+
+/**
+ * A high-level wrapper for Solidity smart contract function calls
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export abstract class BaseDecoder< K extends BaseMethodIdMapping, T extends BaseFunctionCallExplanation, S extends ContractABI > {
+
+  /**
```

Comment:
Just make this an `interface` instead of `abstract class`. The tiny code sharing in constructor is not worth reduction in flexibility for implementations

Reply:
yeah I agree here. `interface` seems like a superior mechanism here